src/System.Data.SqlClient/src/System/Data/SqlClient/SNI/SslOverTdsStream.cs

```diff
@@ -0,0 +1,212 @@
+                    {
+                        currentCount = count;
+                    }
+
+                    count -= currentCount;
+
+                    byte[] header = new byte[8];
+
+                    // We can only send 4088 bytes in one packet. Header[1] is set to 1 if this is a 
+                    // partial packet (whether or not count != 0).
+                    // 
+                    header[0] = 0x12;
+                    header[1] = (byte)(count > 0 ? 0 : 1);
+                    header[2] = (byte)((currentCount + 8) / 256);
+                    header[3] = (byte)((currentCount + 8) % 256);
+                    header[4] = 0;
+                    header[5] = 0;
+                    header[6] = 0;
+                    header[7] = 0;
+
+                    _stream.Write(header, 0, 8);
+                    _stream.Flush();
+                }
+                else
+                {
+                    currentCount = count;
+                    count = 0;
+                }
+
+                _stream.Write(buffer, currentOffset, currentCount);
+                _stream.Flush();
+                currentOffset += currentCount;
+            }
+        }
+
+        /// <summary>
+        /// Set stream length. 
+        /// </summary>
+        /// <param name="value">Length</param>
+        public override void SetLength(long value)
+        {
+            throw NotImplemented.ByDesignWithMessage("Not required");
+        }
+
+        /// <summary>
+        /// Flush stream
+        /// </summary>
+        public override void Flush()
+        {
+            _stream.Flush();
+        }
+
+        /// <summary>
+        /// Get/set stream position
+        /// </summary>
+        public override long Position
+        {
+            get
+            {
+                throw NotImplemented.ByDesignWithMessage("Not required");
+            }
+            set
+            {
+                throw NotImplemented.ByDesignWithMessage("Not required");
+            }
+        }
+
+        /// <summary>
+        /// Seek in stream
+        /// </summary>
+        /// <param name="offset">Offset</param>
+        /// <param name="origin">Origin</param>
+        /// <returns>Position</returns>
+        public override long Seek(long offset, SeekOrigin origin)
+        {
+            throw NotImplemented.ByDesignWithMessage("Not required");
+        }
+
+        /// <summary>
+        /// Check if stream can be read from
+        /// </summary>
+        public override bool CanRead
+        {
+            get { return _stream.CanRead; }
+        }
+
+        /// <summary>
+        /// Check if stream can be written to
+        /// </summary>
+        public override bool CanWrite
+        {
+            get { return _stream.CanWrite; }
+        }
+
+        /// <summary>
+        /// Check if stream can be seeked
+        /// </summary>
+        public override bool CanSeek
+        {
+            get { return _stream.CanSeek; }
```

Comment:
Since we're throwing from all of the seeking-related members, this should probably just return false.